### PR TITLE
evaluator: move the evaluate of ast function date_add and date_sub to…

### PR DESCRIPTION
**Pull request by zxylvlp**
Move the evaluate of ast function date_add and date_sub to builtin.

### CODE REVIEW

**Comment by shenli** [comment]
LGTM

File: evaluator/builtin_time.go [hunk 1]
```diff
@@ -381,3 +385,97 @@ func checkFsp(arg interface{}) (int, error) {
 	}
 	return int(fsp), nil
 }
+
+func builtinDateArith(args []interface{}, ctx context.Context) (interface{}, error) {
+	//// Op is used for distinguishing date_add and date_sub.
```
Comment (zimulala) [inline evaluator/builtin_time.go]: //// -> // ?

**Comment by zxylvlp** [comment]
@zimulala  PTAL

File: evaluator/evaluator_test.go [hunk 2]
```diff
@@ -560,18 +560,28 @@ func (s *testEvaluatorSuite) TestDateArith(c *C) {
...
+				date,
+				dateArithInterval,
 			},
 		}
 		v, err := Eval(ctx, expr)
 		if t.error == true {
 			c.Assert(err, NotNil)
 		} else {
+			fmt.Println(t)
+			fmt.Println(err)
```
Comment (zimulala) [inline evaluator/evaluator_test.go]: can remove these two lines ？

**Comment by zxylvlp** [comment]
@zimulala PTAAL

**Comment by zimulala** [comment]
LGTM

**Pull request merged**